kernel/src/process.rs

```diff
@@ -330,15 +330,56 @@ impl<'a> Process<'a> {
         if data_len.count_ones() != 1 {
             panic!("Tock MPU does not currently handle complex region sizes");
         }
-        let data_region_len = math::log_base_two(data_len as u32);
+        let data_region_len = math::PowerOfTwo::from(data_len as u32);
```

Comment:
nit: You flipped the order below, so now this reads:
data
text
text
data

which is a bit confusing; consider flipping this order too to match